### ISSUE TITLE
Performance issues - nengo_dl backend slower than nengo backend

### ISSUE THREAD

**Issue by loic001** [nengo/nengo-dl#172]
First of all, thank you for this great work!

I tried to run this piece of code in Nengo GUI with the nengo_dl backend (nengo -b nengo_dl). The maximum simulation speed is about 0.11x while with the Nengo backend (nengo -b nengo) I can go up to 4x. Is this normal?

```
#in GUI
import nengo
import numpy as np

model = nengo.Network()
with model as net:
    inp = nengo.Node(output=np.sin)
    ens = nengo.Ensemble(50, 1, neuron_type=nengo.LIF())
    nengo.Connection(inp, ens, synapse=0.1)
    p = nengo.Probe(ens)
```

I also tested outside the GUI by simulating 100s and it's also slower (twice as slow). Where to start to understand where it can come from ?

```
import nengo
import nengo_dl
import numpy as np

with nengo.Network() as net:
    inp = nengo.Node(output=np.sin)
    ens = nengo.Ensemble(50, 1, neuron_type=nengo.LIF())
    nengo.Connection(inp, ens, synapse=0.1)
    p = nengo.Probe(ens)
    
with nengo_dl.Simulator(net, device="GPU") as sim: # this is the only line that changes
    sim.run(100.0)

#simulation time with nengo_dl.Simulator -> about 22s
#simulation time with nengo.Simulator -> about 11s
```

Maybe a problem with the Nvidia drivers? So, I launched the performance tests with the following command:

`python -m pytest -rs --verbose --device gpu --performance nengo_dl/tests/test_benchmarks.py`

Tests are all ok (I had to modify the lower time limit (min arg) because it seems that I have paradoxically better performance than the config used for the tests, see below from file 'test_benchmarks.py'). Anyway, it doesn't exceed the maximum time limit (max arg) specified in file 'test_benchmarks.py'.

```
def test_performance(net, train, minibatch_size, eager, min, max):
    # performance is based on Azure NC6 VM
    # CPU: Intel Xeon E5-2690 v3 @ 2.60Ghz
    # GPU: Nvidia Tesla K80
    # Python version: 3.6.10
    # TensorFlow GPU version: 2.3.0
    # Nengo version: 3.1.0
    # NengoDL version: 3.3.0
```

In short, it's quite strange because the tests seem good, but in practice it's not as good as the nengo backend. Note, apart from these performance issues, everything works (I was able to convert a Keras model into a Nengo network for example). So I wonder if there's really a problem, but still, it would be weird if it was slower on the GPU, wouldn't it?

* GPU: Quadro RTX 6000
* Python version: 3.6.9
* TensorFlow GPU version: 2.3.0
* Nengo version: 3.0.0
* NengoDL version: 3.3.1.dev0
* NVIDIA driver version 450.57
* CUDA Version 11.0

**Comment by drasmuss** [comment]
The basic issue is that these models are too small to show any benefit of running on the GPU. Whenever you run things on the GPU there is a fixed amount of overhead associated with starting up each op (known as the kernel launch time/kernel launch overhead). If the actual GPU execution time is very small, then the launch time will dominate the run time, and you'll end up being slower overall. 

If you switch it to run on the CPU (`nengo_dl.Simulator(net, device="CPU") `), then you should see roughly the same performance in nengo core vs DL. 

Alternatively, if you make the model larger then there will be a point where the GPU performance catches up and overtakes the CPU performance (the exact crossover point will depend on the relative performance of your CPU/GPU).

**Comment by loic001** [comment]
Thank you! Indeed, I have increased the number of neurons/ensembles and it now takes advantage of the GPU.

Note that, according to my tests, the CPU performances with the NengoDL backend are not the same as with Nengo core?...
